include/fiction/algorithms/simulation/sidb/exhaustive_ground_state_simulation.hpp

```diff
@@ -57,15 +31,18 @@ struct exgs_stats
 template <typename Lyt>
 void exhaustive_ground_state_simulation(const Lyt&                        lyt,
                                         const sidb_simulation_parameters& params = sidb_simulation_parameters{},
-                                        exgs_stats<Lyt>*                  ps     = nullptr) noexcept
+                                        sidb_simulation_result<Lyt>*      ps     = nullptr) noexcept
```

Comment:
Ah I see there is a misunderstanding. Sorry for the confusion, I try to be clearer.

The idea of `sidb_simulation_result` is indeed to have a unified interface for all simulators to exchange respective results. I would simply see it as a return type instead of `void` rather than replacing the `..._stats` pointer. If you don't need any additional stats on top of the `sidb_simulation_result`, please simply remove the stats from the function signature.

Reply:
ok, now I got it. That's a great idea! Thanks!